gazebo still generate wrong cmake file

## Issue description
Package that depends on gazebo fails finding it using cmake. (latest master)

## Technical details

Piece of gazebo cmake file:

```
set(GAZEBO_PLUGIN_PATH "/nix/store/9bavxdfavww2araaa9ykx60lxyvkv88z-gazebo-7.0.0//nix/store/9bavxdfavww2araaa9ykx60lxyvkv88z-gazebo-7.0.0/lib/gazebo-7/plugins")

# The media path contains the location on disk where images,
# materials scripts, shaders, and other related resources are stored.
set(GAZEBO_MEDIA_PATH "/nix/store/9bavxdfavww2araaa9ykx60lxyvkv88z-gazebo-7.0.0/share/gazebo-7/media")

set(GAZEBO_PLUGIN_PATH "/nix/store/9bavxdfavww2araaa9ykx60lxyvkv88z-gazebo-7.0.0//nix/store/9bavxdfavww2araaa9ykx60lxyvkv88z-gazebo-7.0.0/lib/gazebo-7.0/plugins")

# The media path contains the location on disk where images,
# materials scripts, shaders, and other related resources are stored.
set(GAZEBO_MEDIA_PATH "/nix/store/9bavxdfavww2araaa9ykx60lxyvkv88z-gazebo-7.0.0/share/gazebo-7.0/media")
```

@Mic92 yes, sorry for the extremely bad bug report. I'm in a hurry and recompiling gazebo takes hours.

This is one a dependency I have in the nix shell I need to run: [default.nix](https://gist.github.com/nico202/90f832713e07ea3fba5bc35dcb1e2eee)

It is failing to find gazebo. This is the error I get:
```
-- Found Protobuf: /nix/store/5s149wk0b1yqkas54r53yzq4lqd5d13v-protobuf-2.6.1-lib/lib/libprotobuf.so;-lpthread (found version "2.6.1") 
CMake Error at /nix/store/i3a69xr2irahb3ykkwrvfxkza6y7an6h-gazebo-6.5.1/lib/cmake/gazebo/gazebo-config.cmake:95 (message):
  Library 'gazebo' in package GAZEBO is not installed properly
Call Stack (most recent call first):
  CMakeLists.txt:14 (find_package)
```

(different gazebo version (6.5/7) because I'm trying both)

--- a/pkgs/applications/science/robotics/gazebo/default.nix
+++ b/pkgs/applications/science/robotics/gazebo/default.nix
@@ -31,7 +31,9 @@ stdenv.mkDerivation rec {
   };
 
   enableParallelBuilding = true; # gazebo needs this so bad
-  cmakeFlags = []
+  cmakeFlags = [
+  "-DCMAKE_INSTALL_LIBDIR:PATH=lib"
+  "-DCMAKE_INSTALL_INCLUDEDIR=include" ]
     ++ optional withQuickBuild [ "-DENABLE_TESTS_COMPILATION=False" ]
     ++ optional withLowMemorySupport [ "-DUSE_LOW_MEMORY_TESTS=True" ]
     ++ optional withHeadless [ "-DENABLE_SCREEN_TESTS=False" ];

